casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -29,6 +33,23 @@ object DeployRuntime {
   def showBlocks[F[_]: Monad: ErrorHandler: Capture: DeployService](): F[Unit] =
     gracefulExit(DeployService[F].showBlocks.map(println(_)))
 
+  def listenForDataAtName[F[_]: Sync: DeployService: Timer: Capture](name: Name): F[Unit] =
+    gracefulExit {
+      ListenAtName.listenAtNameUntilChanges(name) { par: Par =>
+        val request = Channel(ChannelInstance.Quote(par))
+        DeployService[F].listenForDataAtName(request) map (_.blockResults)
+      }
+    }
+
+  def listenForContinuationAtName[F[_]: Sync: Timer: DeployService: Capture](name: Name): F[Unit] =
+    gracefulExit {
+      ListenAtName.listenAtNameUntilChanges(name) { par =>
+        val channel = Channel(ChannelInstance.Quote(par))
+        val request = Channels(Seq(channel))
+        DeployService[F].listenForContinuationAtName(request) map (_.blockResults)
```

Comment:
Or well the user can pass in multiple names

Reply:
Then the same applied for the other API call?

Reply:
I am commenting for the sake of API consistency.

Reply:
> Continuations can be at joins if channels. In other words, we need to return "pars" and then pars.map(par => Channel(Quote(par))) etc.

That makes sense, thanks.

Reply:
Hm, I'm thinking now @KentShikama . If you listen on the join of names then it will return only if the value on both channel has changed right? This might be tricky to implement by hand. I would go with `Observable`s @DoSOfRedRiver for implementing the logic.

Reply:
Ignore my previous comment. When listening for the continuation the change is atomic for all the channels. I was apparently thinking about listening for data on join of channels which doesn't make sense.